search-service/src/main/kotlin/com/egm/stellio/search/csr/service/ContextSourceRegistrationService.kt

```diff
@@ -0,0 +1,227 @@
+            """.trimIndent()
+        ).bind("id", id)
+            .oneToResult { toBoolean(it["exists"]) }
+            .flatMap {
+                if (it && !inverse)
+                    Unit.right()
+                else if (!it && inverse)
+                    Unit.right()
+                else if (it)
+                    AlreadyExistsException(entityAlreadyExistsMessage(id.toString())).left()
+                else
+                    ResourceNotFoundException(entityNotFoundMessage(id.toString())).left()
+            }
+
+    suspend fun getById(id: URI): Either<APIException, ContextSourceRegistration> = either {
+        checkExistence(id).bind()
+
+        val selectStatement =
+            """
+            SELECT id,
+                endpoint,
+                mode,
+                information,
+                operations,
+                observation_interval_start,
+                observation_interval_end,
+                management_interval_start,
+                management_interval_end,
+                created_at,
+                modified_at
+            FROM context_source_registration  
+            WHERE context_source_registration.id = :id
+            """.trimIndent()
+
+        return databaseClient.sql(selectStatement)
+            .bind("id", id)
+            .allToMappedList { rowToContextSourceRegistration(it) }
+            .first().right()
+    }
+
+    suspend fun isCreatorOf(id: URI, sub: Option<Sub>): Either<APIException, Boolean> {
+        val selectStatement =
+            """
+            SELECT sub
+            FROM context_source_registration
+            WHERE id = :id
+            """.trimIndent()
+
+        return databaseClient.sql(selectStatement)
+            .bind("id", id)
+            .oneToResult {
+                it["sub"] == sub.toStringValue()
+            }
+    }
+
+    suspend fun delete(id: URI): Either<APIException, Unit> = either {
+        checkExistence(id).bind()
+        r2dbcEntityTemplate.delete(ContextSourceRegistration::class.java)
+            .matching(query(where("id").`is`(id)))
+            .execute()
+    }
+
+    suspend fun getContextSourceRegistrations(
+        limit: Int,
+        offset: Int,
+        sub: Option<Sub>
+    ): List<ContextSourceRegistration> {
+        val selectStatement =
+            """
+            SELECT id,
+                endpoint,
+                mode,
+                information,
+                operations,
+                observation_interval_start,
+                observation_interval_end,
+                management_interval_start,
+                management_interval_end,
+                created_at,
+                modified_at
+            FROM context_source_registration 
+            WHERE id in (
+                SELECT id 
+                from context_source_registration
+                WHERE context_source_registration.sub = :sub 
+                ORDER BY id
+                limit :limit
+                offset :offset)
```

Comment:
I don't think you need this inner `where ... in` here (it is needed for subscriptions because of the joined tables, you don't have ones here)

Reply:
I thought it was to filter without impacting the result ordering